### DISCUSSION TITLE
Enter password for any action

### DISCUSSION THREAD

**Discussion by homeandriy** [deployphp/deployer#3679] [Help needed]
Hello

I don`t undestand how it works. And have many questions(

First of them, Why do I need to enter the password from the user every time?

Config:
`<?php
namespace Deployer;

require 'recipe/laravel.php';

// Config

set('default_stage', 'staging');
set('application', 'Laravel Test');
set('repository', 'https://github.com/some-git-project.git');

set('branch', 'main');
set('git_tty', false);

//add('shared_files', []);
//add('shared_dirs', []);
//add('writable_dirs', []);

// Hosts

host('xxx.xxx.126.xxx')
    ->setHostname('xxx.xxx.126.xxx')
    ->setRemoteUser('root')
    ->setDeployPath('/www/wwwroot/site-folder.com.ua')
    ->set ('ssh_multiplexing', false);

// Hooks

after('deploy:failed', 'deploy:unlock');
`

And console output (all where out "password", I entered password  for server for ssh):

`$ vendor/bin/dep deploy
task deploy:info
[xxx.xxx.xxx.xxx] info deploying main
task deploy:setup
  ⠋ andriy@xxx.xxx.xxx.xxx's password:   ⠏ 
  ⠏ andriy@xxx.xxx.xxx.xxx's password:   ⠋ 
task deploy:lock
  ⠙ andriy@xxx.xxx.xxx.xxx's password:   ⠧ 
[xxx.xxx.xxx.xxx]  error  in lock.php on line 9:
[xxx.xxx.xxx.xxx] run [ -f /www/wwwroot/site-folder.com.ua/.dep/deploy.lock ] && echo +locked || echo "homeandriy" > /www/wwwroot/site-folder.com.ua/.dep/deploy.lock
[xxx.xxx.xxx.xxx] err bash: line 1: /www/wwwroot/site-folder.com.ua/.dep/deploy.lock: Permission denied
[xxx.xxx.xxx.xxx] exit code 1 (General error)
ERROR: Task deploy:lock failed!
task deploy:failed
task deploy:unlock
  ⠇ andriy@xxx.xxx.xxx.xxx's password:   ⠴ 
    
homeandriy@DESKTOP-04FRSDO-WINDOWS MINGW64 ~/Desktop/Data/project/example-app (main)
$ vendor/bin/dep deploy
task deploy:info
[xxx.xxx.xxx.xxx] info deploying main
task deploy:setup
  ⠼ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠋ @xxx.xxx.xxx.xxx's password:   ⠹ 
task deploy:lock
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠏ 
task deploy:release
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠋ 
  ⠹ @xxx.xxx.xxx.xxx's password:   ⠙ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠹ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠋ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠧ 
task deploy:update_code
  ⠼ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠹ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠋ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠙ 
task deploy:shared
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠼ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠋ @xxx.xxx.xxx.xxx's password:   ⠹ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠏ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠹ @xxx.xxx.xxx.xxx's password:   ⠙ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠋ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠼ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠹ @xxx.xxx.xxx.xxx's password:   ⠸ 
task deploy:writable
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠋ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠹ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠏ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠋ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠙ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠼ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠏ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠋ @xxx.xxx.xxx.xxx's password:   ⠦ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠏ 
  ⠋ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠋ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠹ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠏ @xxx.xxx.xxx.xxx's password:   ⠏ 
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠧ 
  ⠸ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠹ 
  ⠹ @xxx.xxx.xxx.xxx's password:   ⠴ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠋ 
task deploy:vendors
  ⠦ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠙ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠴ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠇ @xxx.xxx.xxx.xxx's password:   ⠸ 
  ⠧ @xxx.xxx.xxx.xxx's password:   ⠇ 
  ⠹ @xxx.xxx.xxx.xxx's password:   ⠼ 
[xxx.xxx.xxx.xxx]  error  in vendors.php on line 31:
[xxx.xxx.xxx.xxx] run cd /www/wwwroot/site-folder.com.ua/releases/1 && /usr/bin/php /usr/bin/composer install --verbose --prefer-dist --no-progress --no-interaction --no-dev --optimize-autoloader 2>&1
[xxx.xxx.xxx.xxx] Do not run Composer as root/super user! See https://getcomposer.org/root for details
[xxx.xxx.xxx.xxx] Installing dependencies from lock file
[xxx.xxx.xxx.xxx] Verifying lock file contents can be installed on current platform.
[xxx.xxx.xxx.xxx] Dependency resolution completed in 0.000 seconds
[xxx.xxx.xxx.xxx] Your lock file does not contain a compatible set of packages. Please run composer update.
[xxx.xxx.xxx.xxx] Problem 1
[xxx.xxx.xxx.xxx] - Root composer.json requires php ^8.0.2 but your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 2
[xxx.xxx.xxx.xxx] - brick/math is locked to version 0.11.0 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - brick/math 0.11.0 requires php ^8.0 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 3
[xxx.xxx.xxx.xxx] - laravel/framework is locked to version v9.52.15 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - laravel/framework v9.52.15 requires php ^8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 4
[xxx.xxx.xxx.xxx] - laravel/sanctum is locked to version v3.2.5 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - laravel/sanctum v3.2.5 requires php ^8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 5
[xxx.xxx.xxx.xxx] - league/flysystem is locked to version 3.15.1 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - league/flysystem 3.15.1 requires php ^8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 6
[xxx.xxx.xxx.xxx] - league/flysystem-local is locked to version 3.15.0 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - league/flysystem-local 3.15.0 requires ext-fileinfo * -> it is missing from your system. Install or enable PHP's fileinfo extension.
[xxx.xxx.xxx.xxx] Problem 7
[xxx.xxx.xxx.xxx] - league/mime-type-detection is locked to version 1.13.0 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - league/mime-type-detection 1.13.0 requires ext-fileinfo * -> it is missing from your system. Install or enable PHP's fileinfo extension.
[xxx.xxx.xxx.xxx] Problem 8
[xxx.xxx.xxx.xxx] - nette/utils is locked to version v4.0.1 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - nette/utils v4.0.1 requires php >=8.0 <8.4 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 9
[xxx.xxx.xxx.xxx] - nunomaduro/termwind is locked to version v1.15.1 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - nunomaduro/termwind v1.15.1 requires php ^8.0 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 10
[xxx.xxx.xxx.xxx] - psr/log is locked to version 3.0.0 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - psr/log 3.0.0 requires php >=8.0.0 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 11
[xxx.xxx.xxx.xxx] - psr/simple-cache is locked to version 3.0.0 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - psr/simple-cache 3.0.0 requires php >=8.0.0 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 12
[xxx.xxx.xxx.xxx] - ramsey/uuid is locked to version 4.7.4 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - ramsey/uuid 4.7.4 requires php ^8.0 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 13
[xxx.xxx.xxx.xxx] - symfony/console is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/console v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 14
[xxx.xxx.xxx.xxx] - symfony/css-selector is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/css-selector v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 15
[xxx.xxx.xxx.xxx] - symfony/deprecation-contracts is locked to version v3.0.2 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/deprecation-contracts v3.0.2 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 16
[xxx.xxx.xxx.xxx] - symfony/error-handler is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/error-handler v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 17
[xxx.xxx.xxx.xxx] - symfony/event-dispatcher is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/event-dispatcher v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 18
[xxx.xxx.xxx.xxx] - symfony/event-dispatcher-contracts is locked to version v3.0.2 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/event-dispatcher-contracts v3.0.2 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 19
[xxx.xxx.xxx.xxx] - symfony/finder is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/finder v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 20
[xxx.xxx.xxx.xxx] - symfony/http-foundation is locked to version v6.0.20 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/http-foundation v6.0.20 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 21
[xxx.xxx.xxx.xxx] - symfony/http-kernel is locked to version v6.0.20 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/http-kernel v6.0.20 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 22
[xxx.xxx.xxx.xxx] - symfony/mailer is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/mailer v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 23
[xxx.xxx.xxx.xxx] - symfony/mime is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/mime v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 24
[xxx.xxx.xxx.xxx] - symfony/process is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/process v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 25
[xxx.xxx.xxx.xxx] - symfony/routing is locked to version v6.0.19 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] - symfony/var-dumper v6.0.19 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] Problem 32
[xxx.xxx.xxx.xxx] - symfony/http-foundation v6.0.20 requires php >=8.0.2 -> your php version (7.4.33) does not satisfy that requirement.
[xxx.xxx.xxx.xxx] - fruitcake/php-cors v1.2.0 requires symfony/http-foundation ^4.4|^5.4|^6 -> satisfiable by symfony/http-foundation[v6.0.20].
[xxx.xxx.xxx.xxx] - fruitcake/php-cors is locked to version v1.2.0 and an update of this package was not requested.
[xxx.xxx.xxx.xxx] To enable extensions, verify that they are enabled in your .ini files:
[xxx.xxx.xxx.xxx] - /www/server/php/74/etc/php-cli.ini
[xxx.xxx.xxx.xxx] You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.
[xxx.xxx.xxx.xxx] exit code 2 (Misuse of shell builtins)
ERROR: Task deploy:vendors failed!
task deploy:failed
task deploy:unlock
  ⠼ @xxx.xxx.xxx.xxx's password:   ⠙`

**Comment by acharseth** [comment]
Having to enter password has nothing to do with Deployer. You can setup a credential manager & ssh agent to remember the password for you, run an initial ssh connection to make the shell remember the password (eg. `ssh fetch`) and then you can run your Deployer script without having to enter a password. Or you could have your script ask for the password in the start of the script, store it in a variable and use it for every SSH-command in Deployer.
